packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -179,13 +183,11 @@ export function PrivateBlockToolbar( {
 			key={ toolbarKey }
 		>
 			<div ref={ toolbarWrapperRef } className={ innerClasses }>
-				{ ! isMultiToolbar &&
-					isLargeViewport &&
-					isDefaultEditingMode && <BlockParentSelector /> }
+				{ showParentSelector && ! isMultiToolbar && isLargeViewport && (
+					<BlockParentSelector />
+				) }
 				{ ( shouldShowVisualToolbar || isMultiToolbar ) &&
-					( isDefaultEditingMode ||
-						( isContentOnlyEditingMode && ! hasParentPattern ) ||
-						isSynced ) && (
+					! hasParentPattern && (
```

Comment:
Should the `isDefaultEditingMode` and `isContentOnlyEditingMode` checks be removed here? That change seems to be part of https://github.com/WordPress/gutenberg/pull/65204 that wasn't being backported.

Reply:
> In order for the block toolbar to work properly, I had to include in the backport a small chunk of this PR #65485 as otherwise the toolbar doesn't appear in blocks with non default mode (which is the case for zoom-out sections)

Ah, this must be what you were talking about [here](https://github.com/WordPress/gutenberg/pull/66200#issuecomment-2419233966).

Reply:
I think we need to test the toolbar in zoom-out (it should appear) and we should check whether the toolbar in "content-only" blocks is unchanged/makes sense.